### PR TITLE
Filepicker Improvement - Don't store large files in the dataStore

### LINKED ISSUE #6297: [Bug] Filepicker Crashes with Large Files
## Description

A user reported that the file picker crashes while uploading files > 10 mb. We triaged this is due to the file getting stored in the data tree.

The solution is to not load files > 5mb into the data tree from the file picker

### PR DESCRIPTION
**Pull request by SatishGandham**
## Description
Solves the issue of file picker crashing on large file uploads by not storing files larger than 5MB in dataTree.

Fixes #6297 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: bug-fix/dont-stire-large-files-in-dataTree 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 53.25 **(-0.04)** | 34.72 **(-0.04)** | 31.32 **(-0.02)** | 53.82 **(-0.03)**
 :red_circle: | app/client/src/sagas/ActionExecutionSagas.ts | 12.53 **(-0.24)** | 1.93 **(-0.07)** | 8.82 **(-0.86)** | 14.47 **(-0.13)**
 :red_circle: | app/client/src/utils/AppsmithUtils.tsx | 53.44 **(-0.23)** | 26.14 **(-0.6)** | 32.5 **(-2.64)** | 48.15 **(-0.87)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.71 **(0.24)** | 38.6 **(0.88)** | 36.21 **(0)** | 55.35 **(0.27)**
 :red_circle: | app/client/src/widgets/FilepickerWidget.tsx | 35.24 **(-3.22)** | 0 **(0)** | 17.24 **(-0.62)** | 35.64 **(-2.14)**</details>

### CODE REVIEW

**Review by rahulramesha — APPROVED** [review]
Approving for tests

**Review by jsartisan — APPROVED** [review]

File: app/client/src/widgets/FilepickerWidget.tsx [hunk 1]
```diff
@@ -341,39 +342,46 @@ class FilePickerWidget extends BaseWidget<
...
+              reader.readAsBinaryString(file.data);
+            } else {
+              reader.readAsText(file.data);
+            }
+            reader.onloadend = () => {
+              const newFile = {
+                type: file.type,
+                id: file.id,
+                data: reader.result,
+                name: file.meta ? file.meta.name : undefined,
```
Comment (hetunandu) [inline app/client/src/widgets/FilepickerWidget.tsx:367]: Set a default name instead of undefined

**Review by hetunandu — APPROVED** [review]

**Comment by SatishGandham** [comment]
#6692 created.

**Pull request closed without merging**